packages/core/src/Machine.ts

```diff
@@ -55,15 +55,17 @@ export function createMachine<
   TEvent extends EventObject = ModelEventsFrom<TModel>,
   TTypestate extends Typestate<TContext> = { value: any; context: TContext }
 >(
-  config: MachineConfig<TContext, any, TEvent>,
+  config: MachineConfig<TContext, any, TEvent> & { context: TContext },
   options?: Partial<MachineOptions<TContext, TEvent>>
 ): StateMachine<TContext, any, TEvent, TTypestate>;
 export function createMachine<
   TContext,
   TEvent extends EventObject = AnyEventObject,
   TTypestate extends Typestate<TContext> = { value: any; context: TContext }
 >(
-  config: MachineConfig<TContext, any, TEvent>,
+  config: TContext extends Model<any, any, any>
+    ? never
+    : MachineConfig<TContext, any, TEvent>,
```

Comment:
This is supposed to help with cryptic errors when we fail to match the first overload and Model ges used in the second one as TContext, right? maybe worth adding a comment about this here?

Reply:
Yes, will do. The main use-case is that this:

```ts
createMachine<typeof model>({
  // no context
});
```

Will go to the second overload and assume that the `typeof model` is actually the `TContext`, which is bad.

Reply:
8a1587e